build-system/compile/json-schema/index.js

```diff
@@ -182,42 +182,209 @@ function transformAjvCode(code, scope, config) {
   const findProperty = (properties, name) =>
     properties.find((node) => isObjectProperty(node, name));
 
+  /**
+   * @param {babel.NodePath<babel.types.MemberExpression>} memberExpression
+   * @return {null | undefined | number | string}
+   */
+  function evaluatePropertyKey(memberExpression) {
+    const property = memberExpression.get('property');
+    return property.isIdentifier()
+      ? property.node.name
+      : property.evaluate().value;
+  }
+
+  const schemaIdRegexp = /^schema[0-9]*$/;
+
+  /**
+   * @param {babel.NodePath<any>} path
+   * @param {string[]} keys
+   * @return {boolean}
+   */
+  function isMemberExpressionLeftwards(path, keys) {
+    let i = keys.length - 1;
+    while (true) {
+      if (path.isMemberExpression()) {
+        const key = evaluatePropertyKey(path);
+        if (key !== keys[i--]) {
+          return false;
+        }
+        path = path.get('object');
+      } else if (path.isVariableDeclarator()) {
+        const init = path.get('init');
+        if (!init) {
+          break;
+        }
+        path = init;
+      } else if (path.isIdentifier()) {
+        const binding = path.scope.getBinding(path.node.name);
+        if (!binding) {
+          break;
+        }
+        path = binding.path;
+      } else {
+        break;
+      }
+    }
+    return i === 0 && path.isIdentifier({name: keys[i]});
+  }
+
+  /**
+   * @param {babel.NodePath} path
+   * @return {boolean}
+   */
+  function isReferenceToObjectKeyList(path) {
+    const {parentPath} = path;
+    if (parentPath?.isMemberExpression() && path.key === 'object') {
```

Comment:
```suggestion
    if (parentPath?.isMemberExpression({ key: path.node })) {
```

Reply:
`key` is a property of `NodePath`, not `Node`.